lib/puppet/provider/grub_config/grub2.rb

```diff
@@ -66,6 +66,12 @@ def value
   end
 
   def value=(newval)
+    if newval.is_a?(String)
+      unless %w[' "].include?(newval[0].chr)
+        newval = %Q("#{newval}")
```

Comment:
I don't understand what you're doing here

Reply:
@raphink Ensuring that we don't quote strings that already have quotes in them.

Reply:
Oh I see, `%Q("")` adds literal quotes while interpreting `#{newval}`.

Reply:
iirc, there's a a `quote_it` function in augeasproviders_core, which is used in the shellvar module.